### PR TITLE
Feature/mixedbread ai integration

### PR DESCRIPTION
**Pull request by juliuslipp**
- [x] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** Add support for mixedbread-ai's embeddding & reranking API.
    - **Dependencies:** mixedbread-ai="^2.2.6"
    - **Twitter handle:** mixedbreadai

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

**Comment by juliuslipp** [comment]
Hey guys! @huangrpablo  added this integration for [mixedbread](https://mixedbread.ai).  With mixedbread we're building SOTA models and tools to streamline retrieval. So far, we trained some of the most widely used open-source embeddings and reranking models (https://huggingface.co/mixedbread-ai).

I don't know if you can still remember embaas @hwchase17 , as one of the first embedding providers available on langchain. Would love to partner up and push this a little bit. Lmk what you think!!

We are deprecating the old embaas package as well :)

File: libs/partners/mixedbreadai/docs/rerank.ipynb
Comment (baskaryan) [inline libs/partners/mixedbreadai/docs/rerank.ipynb:1]: should be moved to top level /docs/docs/integrations/document_transformers/

File: libs/partners/mixedbreadai/docs/text_embedding.ipynb
Comment (baskaryan) [inline libs/partners/mixedbreadai/docs/text_embedding.ipynb:1]: should be moved to top level /docs/docs/integrations/text_embedding/

File: libs/community/langchain_community/embeddings/embaas.py [hunk 1]
```diff
@@ -20,6 +21,11 @@ class EmbaasEmbeddingsPayload(TypedDict):
     instruction: NotRequired[str]
 
 
+@deprecated(
+    since="0.2.5",
```
Comment (baskaryan) [inline libs/community/langchain_community/embeddings/embaas.py:25]: 0.2.6 (the next version)

**Comment by juliuslipp** [comment]
@baskaryan done!

**Comment by huangrpablo** [comment]
@baskaryan hey, may I get the CI workflows approved?

**Comment by juliuslipp** [comment]
@efriis @baskaryan any updates?

**Comment by efriis** [comment]
Hey there! Sorry for the delay. I don't think we've been in contact with your team about partnering on this, so I'll close it (if you have been, please let me know who you're talking to!)

Would recommend adding this to `langchain-community` or self-publishing this as `mixedbread-langchain` and documenting how to use it for now!

**Pull request closed without merging**